src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -11,37 +11,64 @@ import './ActivityGraph.css';
 import type {
   IndexIntoSamplesTable,
   IndexIntoCategoryList,
-  SamplesTable,
-  StackTable,
+  Thread,
 } from '../../../types/profile';
 import type {
   Milliseconds,
   DevicePixels,
   CssPixels,
 } from '../../../types/units';
-import type { ActivityGraphProps } from './ActivityGraph';
+
+/**
+ * Computing the ActivityGraph's fills requires a set of immutable properties that
+ * are used to compute the fills, but are never changed. In order to separate out
+ * the mutable values, collect all of these immutable settings in one object that
+ * can easily be shared between classes and functions.
+ */
+type LastRenderedGraphSettings = {|
```

Comment:
I'd call this just `GraphRenderSettings`. Usually this is a new set of settings. The only place where the "last" aspect comes into play is if you query things from the `fillsQuerier` - it uses information from the previous rendering.

Reply:
```
/**
 * This type contains the values that were used to render the ThreadActivityGraph's React
 * component, plus the sizing of the DOM element. These values are computed BEFORE drawing
 * to the 2d canvas, and are needed to compute the 2d canvas' fills.
 *
 * Computing these fills requires a set of mutable and immutable values. This type helps
 * delineate these two types of values by only containing the immutable values. This
 * object makes it easy to share these values between different classes and functions.
 */
type RenderedComponentSettings = {|
  +canvasPixelWidth: DevicePixels,
  +canvasPixelHeight: DevicePixels,
  +fullThread: Thread,
  +interval: Milliseconds,
  +rangeStart: Milliseconds,
  +rangeEnd: Milliseconds,
  +xPixelsPerMs: number,
  +treeOrderSampleComparator: ?(
    IndexIntoSamplesTable,
    IndexIntoSamplesTable
  ) => number,
  +greyCategoryIndex: IndexIntoCategoryList,
  +samplesSelectedStates: ?Array<boolean>,
  +categoryDrawStyles: CategoryDrawStyles,
|};
```